File: webapps/frontend/app/viewmodels/containers/pmachine.js
// Copyright 2014 CloudFounders NV
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
/*global define */
define([
    'jquery', 'knockout',
    'ovs/generic', 'ovs/api'
], function($, ko, generic, api) {
    "use strict";
    return function(guid) {
        var self = this;

        // Variables
        self.loadHandle = undefined;

        // Observables
        self.loading = ko.observable(false);
        self.loaded  = ko.observable(false);

        self.guid   = ko.observable(guid);
        self.name   = ko.observable();
        self.hvtype = ko.observable();

        self.load = function() {
            return $.Deferred(function(deferred) {
                self.loading(true);
                api.get('pmachines/' + self.guid())
                    .done(function(data) {
                        self.name(data.name);
                        self.hvtype(data.hvtype);

                        self.loaded(true);
                        deferred.resolve();
                    })
                    .fail(deferred.reject)
                    .always(function() {
                        self.loading(false);
                    })
            }).promise();
        }
    };
});

File: ovs/dal/hybrids/pmachine.py
# Copyright 2014 CloudFounders NV
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""
PMachine module
"""
from ovs.dal.dataobject import DataObject


class PMachine(DataObject):
    """
    The PMachine class represents a pMachine. A pMachine is the physical machine
    running the Hypervisor.
    """
    # pylint: disable=line-too-long
    _blueprint = {'name':        (None, str, 'Name of the pMachine.'),
                  'description': (None, str, 'Description of the pMachine.'),
                  'username':    (None, str, 'Username of the pMachine.'),
                  'password':    (None, str, 'Password of the pMachine.'),
                  'ip':          (None, str, 'IP address of the pMachine.'),
                  'hvtype':      (None, ['HYPERV', 'VMWARE', 'XEN', 'KVM'], 'Hypervisor type running on the pMachine.')}
    _relations = {}
    _expiry = {}
    # pylint: enable=line-too-long
